mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -580,10 +581,14 @@ class Word2VecModel private[spark] (
       ind += 1
     }
 
-    val scored = wordList.zip(cosVec).toSeq.sortBy(-_._2)
+    val pq = new BoundedPriorityQueue[(String, Double)](num + 1)(Ordering.by(_._2))
+
+    pq ++= wordList.zip(cosVec)
```

Comment:
I'm OK to merge this as-is as it is an improvement. I know one of the original purposes was to avoid copies. I suppose it's a little more verbose, but avoids a collection copy, to do ...

```
for (i <- cosVec.indices) {
  pq += (wordList(i), cosVec(i))
}
```

I don't feel strongly about it.